edx_event_bus_kafka/internal/producer.py

```diff
@@ -136,26 +136,42 @@ def extract_key_schema(signal_serializer: AvroSignalSerializer, event_key_field:
     return json.dumps(subschema, sort_keys=True)
 
 
+@lru_cache()
+def get_signal_serializer(signal: OpenEdxPublicSignal):
```

Comment:
I think we'll also want to reset this cache in `_reset_caches` at the end of the file.